arbnode/node.go

```diff
@@ -666,7 +666,9 @@ func createNodeImpl(
 		if err := wallet.Initialize(ctx); err != nil {
 			return nil, err
 		}
-		stakerAddr = dp.Sender()
+		if dp != nil {
```

Comment:
dp is also used in validatorwallet.NewContract, validatorwallet.NewEOA - Looking over it a few minutes, I'm not convinced a nil value there won't cause problems